src/vs/workbench/contrib/inlineChat/browser/inlineChatDecorations.ts

```diff
@@ -59,17 +60,30 @@ export class InlineChatDecorationsContribution extends Disposable implements IEd
 			}
 			this._onEnablementOrModelChanged();
 		}));
+		this._register(this._inlineChatSessionService.onWillStartSession(() => this._localToDispose.clear()));
```

Comment:
Always call this._onEnablementOrModelChanged()
Only in there we do `this._localToDispose` and clear all the state.

Reply:
Commented in dm